client/src/index.js

```diff
@@ -17,14 +19,27 @@ const cache = new InMemoryCache({
 });
 
 const httpLink = createHttpLink({
-  uri: "http://localhost:5000/graphql",
-  headers: {
-    authorization: token
+  uri: "http://localhost:5000/graphql"
+});
+
+const authLink = setContext((_, { headers }) => {
+  const token = localStorage.getItem("auth-token");
+  return {
+    headers: {
+      ...headers,
+      authorization: token ? token : ""
+    }
+  };
+});
+
+const errorLink = onError(({ graphQLErrors }) => {
```

Comment:
I'm a little confused. Could you clarify what this function is for? Is the client 'onError' the same as this function?

Reply:
I copied this from what Bryce posted in Slack, I'm not 100% sure whats going on in it but Bryce said that setting it up this way solves refreshing issues or something like that. It seems that `onError` may be handling errors specific to the Apollo link or something?